### ISSUE TITLE
Merged Cells?

### ISSUE THREAD

**Issue by smckissock** [PMSI-AlignAlytics/scrollgrid#3]
We're looking to use this for a replacement of the dc.js dataTable.  A missing feature in both is merged cells, so we can add spanning column headers.  Could you add that?

**Comment by johnkiernander** [comment]
We also need this, so it is something I'll be looking at when I get a chance.

**Comment by nagarajanchinnasamy** [comment]
Yes... This is much needed feature. Will be nice to have it asap :-)

**Comment by johnkiernander** [comment]
I've been giving this a bit of thought and I think the approach I'll use is to add a row span and column span which can be applied through format rules.  As I see it, there are a couple of considerations.

Firstly because we need to consider that adapters can make asynchronous calls to a server to get just the visible data range.  That means that the value for a visible merged cell wouldn't currently be accessible (where the top left cell of the merged area is not in the visible range).  Therefore I'm going to need to extend the getData method of the adapter to handle that.  

Secondly it's going to need to completely ignore cells which are covered by the spanned cell with precedence going to the toppest or leftest cell.  This allows us to specify in the format rules that (for example) row 1 has a column span of 2, then every other cell would be skipped and replaced by the left cell.

I think that will be enough to get it working.  I originally planned to have some kind of logic to group common values but that will have to be handled in the calling code, I can't put that kind of logic in Scrollgrid, mainly due to the first point above.

Does that approach sound like it would work for you both?

**Comment by smckissock** [comment]
Yes, that sounds good. And having it group common values itself is not something we're interested in - too many weird cases - so that's good too.

**Comment by nagarajanchinnasamy** [comment]
Yes. Thats great. Handling data (including headers) should be left to adapters. In addition to controlling it through configuration (formatRules) it will be nice if equivalent API is available to invoke at runtime.

**Comment by nagarajanchinnasamy** [comment]
@johnkiernander in Tcl/Tk (Tk actually), the `grid` command uses a specific syntax to mention the layout. Please check [here...](http://www.tcl.tk/man/tcl8.6/TkCmd/grid.htm#M28)

**Comment by nagarajanchinnasamy** [comment]
Any further development on this?

**Comment by johnkiernander** [comment]
I've spent quite a bit of time on this but I can't get it to work in a way that is performant enough.  There's a separate branch called col-row-span which you can fork and switch to if you need this now, but I can't merge with master until I find a solution which is quick enough with big tables.

**Comment by nagarajanchinnasamy** [comment]
@johnkiernander Thanks for the branch. I plan to use scrollgrid in [subtotal.js](http://nagarajanchinnasamy.com/subtotal/) to render pivot table. That's the main reason I've asked for this feature. I will experiment with the col-row-span branch and let you know my feedback.

Thanks again :)
